packages/react-next/src/components/SwatchColorPicker/ColorPickerGridCell.base.tsx

```diff
@@ -12,146 +12,121 @@ import {
   IColorPickerGridCellStyles,
 } from './ColorPickerGridCell.types';
 
-const getColorPickerGridCellButtonClassNames = memoizeFunction(
-  (
+const getClassNames = classNamesFunction<IColorPickerGridCellStyleProps, IColorPickerGridCellStyles>();
+
+/** Validate if the cell's color is white or not to apply whiteCell style */
+const isWhiteCell = (inputColor: string): boolean => {
+  const currentColor = getColorFromString(inputColor!);
+  return currentColor?.hex === 'ffffff';
+};
+
+export const ColorPickerGridCellBase: React.FunctionComponent<IColorPickerGridCellProps> = props => {
+  const {
+    item,
+    // eslint-disable-next-line deprecation/deprecation
+    idPrefix = props.id,
+    selected = false,
+    disabled = false,
+    styles,
+    circle = true,
+    color,
+    onClick,
+    onHover,
+    onFocus,
+    onMouseEnter,
+    onMouseMove,
+    onMouseLeave,
+    onWheel,
+    onKeyDown,
+    height,
+    width,
+    borderWidth,
+  } = props;
+
+  const classNames: IProcessedStyleSet<IColorPickerGridCellStyles> = getClassNames(styles!, {
+    theme: props.theme!,
+    disabled,
+    selected,
+    circle,
+    isWhite: isWhiteCell(color!),
+    height,
+    width,
+    borderWidth,
+  });
+
+  const getColorPickerGridCellButtonClassNames = (
```

Comment:
Please move this back to the exact previous implementation, where it's outside the component body and memoized. If I remember right that was an important perf optimization for a partner.